pkg/rehearse/executor.go

```diff
@@ -0,0 +1,147 @@
+package rehearse
+
+import (
+	"fmt"
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/sets"
+	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	pj "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+type Result struct {
+	Name   string
+	Status pjapi.ProwJobStatus
+}
+
+type Executor struct {
+	dryRun     bool
+	rehearsals []*prowconfig.Presubmit
+	prNumber   int
+	prRepo     string
+	refs       *pjapi.Refs
+	loggers    Loggers
+	pjclient   pj.ProwJobInterface
+	resultChan chan Result
+	errorChan  chan error
+	quit       chan struct{}
+}
+
+func NewExecutor(toBeRehearsed map[string][]prowconfig.Presubmit, prNumber int, prRepo string, refs *pjapi.Refs,
+	loggers Loggers, pjclient pj.ProwJobInterface, dryRun bool, resultChan chan Result, errorChan chan error, quit chan struct{}) *Executor {
```

Comment:
10 params API is a bit clunky, I would consider coalescing the channels into a struct, they are always used together anyway.

Reply:
This is all package-local so you do not need an exported New method and you can just create a struct literal for the executor